### PR TITLE
[CN-1163] Improve restore E2E tests

### PR DESCRIPTION
**Pull request by semihbkgr**

### CODE REVIEW

File: test/e2e/backup_restore_test.go [hunk 1]
```diff
@@ -505,9 +537,7 @@ var _ = Describe("Hazelcast CR with Persistence feature enabled", Group("backup_
 			Expect(k8sClient.Create(context.Background(), hotBackup)).Should(Succeed())
 			assertHotBackupSuccess(hotBackup, 1*Minute)
 
-			By("filling the cache with entries after backup")
-			fillCachePortForward(hazelcast, cache.GetDSName(), localPort, entryCount)
```
Comment (cagric0) [inline test/e2e/backup_restore_test.go:509]: Why did you remove filling step after backup? Shouldn't we keep it for proper testing. As far as I can understand this test is testing second put shouldn't be appear after restore. Am I missing something?

Comment (semihbkgr) [reply test/e2e/backup_restore_test.go:509]: It is simply a test case for checking the persistency on the caches entries: `should check cache entry persistence after HotBackup`

> As far as I can understand this test is testing second put shouldn't be appear after restore.

I thought that we don't need to include this edge case scenario in the sole test case where we check the cache persistency. We already have multiple test cases examining the behavior of restoring when hot-restart data is present.

Additionally, it's important to note that the `validateCacheEntriesPortForward` test util function is not correctly implemented due to the lack of cache API support in the Go client. Currently, it only verifies the existence of entries in the cache, ignoring the cache size and other entries assumed not to be present in the cache."

**Review by cagric0 — APPROVED** [review]
